### PR TITLE
fix: cjs compact

### PR DESCRIPTION
**Pull request by sxzz**
### Description

`tinypool` is a pure ESM package, so I replace it with `piscina`.

And enables `shims` to shim `import.meta.url` for cjs version.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
       "import": "./dist/index.js"
     }
   },
-  "main": "./dist/index.js",
+  "main": "./dist/index.cjs",
```
Comment (sxzz) [inline package.json]: I'm not sure about `type=module` package, should it point to CJS or ESM version?

**Comment by antfu** [comment]
I prefer to keep using tinypool. If we import via `const Pool = await import('tinypool')` it should be fine for it to be pure ESM

**Comment by sxzz** [comment]
Updated.

File: package.json (hunk 1 above)
Comment (antfu) [reply package.json]: type=module's main should point to ESM

Comment (sxzz) [reply package.json]: Ok, just reverted.

**Pull request merged**